extensions/jdbc/jdbc-mariadb/deployment/src/main/java/io/quarkus/jdbc/mariadb/deployment/JDBCMariaDBProcessor.java

```diff
@@ -25,10 +26,9 @@ FeatureBuildItem feature() {
     }
 
     @BuildStep
-    void registerDriver(BuildProducer<JdbcDriverBuildItem> jdbcDriver,
-            SslNativeConfigBuildItem sslNativeConfigBuildItem) {
-        jdbcDriver.produce(new JdbcDriverBuildItem(DatabaseKind.MARIADB, "org.mariadb.jdbc.Driver",
-                "org.mariadb.jdbc.MySQLDataSource"));
+    void registerDriver(BuildProducer<JdbcDriverBuildItem> jdbcDriver) {
+        jdbcDriver.produce(
+                new JdbcDriverBuildItem(DatabaseKind.MARIADB, "org.mariadb.jdbc.Driver", "org.mariadb.jdbc.MariaDbDataSource"));
```

Comment:
As for the `xaDriverClass` (last param), there is also `MariaDbPoolConnection` but the non-pooling variant felt more logical (given what was there before).

Both classes were introduced to 3.0 via https://github.com/mariadb-corporation/mariadb-connector-j/commit/2999238f